ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift

```diff
@@ -206,6 +196,76 @@ struct AIChatResponseMessageView: View {
 
     return Text(text + period + space + learnMoreLink)
   }
+
+  /// View for displaying a Code Block with a header displaying language hint & copy button,
+  /// line numbers and the highlighted code.
+  struct CodeBlockView: View {
+
+    let block: MarkdownParser.StringBlock
+    let codeBlock: MarkdownParser.CodeBlock
+
+    var body: some View {
+      VStack(spacing: 0) {
+        headerSection
+          .background(Color(braveSystemName: .pageBackground))
+        codeBlockSection
+          .background(codeBlock.backgroundColor)
+      }
+      .clipShape(RoundedRectangle(cornerRadius: 8.0, style: .continuous))
+      .overlay(
+        RoundedRectangle(cornerRadius: 8)
+          .inset(by: 0.5)
+          .stroke(Color(braveSystemName: .dividerSubtle), lineWidth: 1)
+      )
+    }
+
+    private var headerSection: some View {
+      HStack {
+        Text(
+          codeBlock.languageHint?.capitalizeFirstLetter ?? Strings.AIChat.leoCodeExampleDefaultTitle
+        )
+        .foregroundStyle(Color(braveSystemName: .textPrimary))
+        Spacer()
+        Button(
+          action: {
+            UIPasteboard.general.setValue(
+              String(block.string.characters),
+              forPasteboardType: "public.plain-text"
+            )
+          },
+          label: {
+            HStack(spacing: 8) {
+              Image(braveSystemName: "leo.copy")
+              Text(Strings.AIChat.responseContextMenuCopyTitle)
+                .fontWeight(.semibold)
+            }
+            .foregroundStyle(Color(braveSystemName: .textSecondary))
+          }
+        )
+      }
+      .padding(16)
+    }
+
+    private var codeBlockSection: some View {
+      Grid(alignment: .leading, horizontalSpacing: nil, verticalSpacing: nil) {
+        ForEach(Array(block.string.split(separator: "\n").enumerated()), id: \.offset) {
```

Comment:
offset is not a stable identifier and shouldn't be used

Reply:
Updated to unique identifier in [65264be](https://github.com/brave/brave-core/pull/24606/commits/65264be48db73e843ca2ffd6356fb972d3b12930)